### PR TITLE
[Ruby][faraday] fix download_file

### PR DESCRIPTION
**Pull request by tsuwatch**
fixes: #7734

`download_file` doesn't work.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @cliffano (2017/07) @zlx (2017/09) @autopp (2019/02)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache [hunk 1]
```diff
@@ -128,40 +128,11 @@
...
-    # from the "Content-Disposition" header if provided, otherwise a random filename.
-    # The response body is written to the file in chunks in order to handle files which
-    # size is larger than maximum Ruby String or even larger than the maximum memory a Ruby
-    # process can use.
-    #
-    # @see Configuration#temp_folder_path
     def download_file(request)
-      tempfile = nil
-      encoding = nil
-      request.headers do |response|
```
Comment (tsuwatch) [inline modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache:141]: Does not process this block.

File: modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache [hunk 2]
```diff
@@ -128,40 +128,11 @@
...
-        content_disposition = response.headers['Content-Disposition']
-        if content_disposition && content_disposition =~ /filename=/i
-          filename = content_disposition[/filename=['"]?([^'"\s]+)['"]?/, 1]
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache:144]: @tsuwatch thanks for the PR. Would it be possible to keep the filename from the response headers if available?

File: samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb [hunk 3]
```diff
@@ -232,7 +203,8 @@ def deserialize(response, return_type)
       # handle file downloading - return the File instance processed in request callbacks
       # note that response body is empty when the file is written in chunks in request on_body callback
       if return_type == 'File'
-        @tempfile.write(@stream)
+        @tempfile = Tempfile.open('download-', @config.temp_folder_path, encoding: body.encoding)
+        @tempfile.write(@stream.join.force_encoding(body.encoding))
```
Comment (tsuwatch) [inline samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb:207]: `@stream` is array

Comment (tsuwatch) [reply samples/client/petstore/ruby-faraday/lib/petstore/api_client.rb:207]: Need to `force_encoding`

File: modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache (hunk 2 above)
Comment (tsuwatch) [reply modules/openapi-generator/src/main/resources/ruby-client/api_client_faraday_partial.mustache:144]: did it.
https://github.com/OpenAPITools/openapi-generator/pull/7842/commits/3e3ea88b0dec7763c861e9623f88a9d1fa9034c8

**Comment by wing328** [comment]
CI failure not related to this PR.

**Comment by tsuwatch** [comment]
Thanks

**Pull request merged**